mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/MantisAkkaRpcSystemLoader.java

```diff
@@ -16,24 +16,71 @@
 
 package io.mantisrx.server.core;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ServiceLoader;
+import java.util.UUID;
 import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.CoreOptions;
+import org.apache.flink.core.classloading.ComponentClassLoader;
 import org.apache.flink.runtime.rpc.RpcSystem;
 import org.apache.flink.runtime.rpc.RpcSystemLoader;
-import org.apache.flink.runtime.rpc.akka.AkkaRpcSystem;
+import org.apache.flink.util.IOUtils;
 
 /**
  * RpcSystemLoader for mantis task executor and other services that need to expose an RPC API.
  * This particular implementation uses the akka RPC implementation under the hood.
  */
 public class MantisAkkaRpcSystemLoader implements RpcSystemLoader {
 
-    private static final RpcSystem INSTANCE = new AkkaRpcSystem();
+    private static final RpcSystem INSTANCE = createRpcSystem();
+
+    public static RpcSystem getInstance() {
+        return INSTANCE;
+    }
+
     @Override
     public RpcSystem loadRpcSystem(Configuration config) {
         return INSTANCE;
     }
 
-    public static RpcSystem getInstance() {
-        return INSTANCE;
+    private static RpcSystem createRpcSystem() {
+        try {
+            final ClassLoader flinkClassLoader = RpcSystem.class.getClassLoader();
+
+            final Path tmpDirectory = Paths.get(System.getProperty("java.io.tmpdir"));
+            Files.createDirectories(tmpDirectory);
+            final Path tempFile =
+                Files.createFile(
+                    tmpDirectory.resolve("flink-rpc-akka_" + UUID.randomUUID() + ".jar"));
```

Comment:
minor: add a prefix/folder from e.g. jobId. 
minor2: (for your consideration) make this configurable.

Reply:
what'd be the benefit of this?